litellm/llms/solar.py

```diff
@@ -0,0 +1,176 @@
+import os, types
+import json
+from enum import Enum
+import requests
+import time
+from typing import Callable, Optional
+import litellm
+import httpx
+from litellm.utils import ModelResponse, Usage
+from .prompt_templates.factory import prompt_factory, custom_prompt
+
+
+class SolarError(Exception):
```

Comment:
if solar is openai-compatible, a separate class is not needed @Tokkiu 

See the GROQ implementation for referencehttps://github.com/BerriAI/litellm/pull/2168